dvc/repo/live.py

```diff
@@ -1,15 +1,30 @@
 import contextlib
+import logging
 import os
 from typing import TYPE_CHECKING, List, Optional
 
 from dvc.exceptions import MetricDoesNotExistError, MetricsError
 
+logger = logging.getLogger(__name__)
+
 if TYPE_CHECKING:
     from dvc.output import BaseOutput
     from dvc.path_info import PathInfo
     from dvc.repo import Repo
 
 
+def create_summary(out):
+    from dvc.utils.html import write
+
+    assert out.live and out.live["html"]
+
+    metrics, plots = out.repo.live.show(str(out.path_info))
+
+    html_path = out.path_info.with_suffix(".html")
+    write(html_path, plots, metrics)
+    logger.info(f"\nfile://{os.path.abspath(html_path)}")
```

Comment:
It doesn't look like there's much to this HTML summary (either here or in https://github.com/iterative/dvc/blob/master/dvc/utils/html.py). Would it make sense to build out summary capability in dvclive? That would allow for complete dvclive functionality without dvc.

Also, could the summary output format be abstracted in dvclive so that non-HTML outputs could be built? For example, a matplotlib output that auto-updates, or even a very basic cli output. Different output types could enable users to see something like a progress bar for their model performance without opening a separate page that needs to be manually refreshed.

Reply:
@dberenbaum 
This question has few levels.
1.  > Would it make sense to build out summary capability in dvclive? That would allow for complete dvclive functionality without dvc

I guess it would make sense, but it would mean copying not only html module and this particular function, but quite a lot of code from `metrics` and `plots` (`live.show` is actually calling them) - so all the functionality related to vega-js would probably need to be copied too, the templates for example. Also having it in dvc allows us to use a command `dvc live` to visualize the `live` outputs. Its not too flashy as of today, but AFAIK, it is supposed to be iterated upon in the future.

> non-HTML outputs could be built ?

Probably, but that leaves us with a lot of questions related to plots itself - how would it work on different os-es? What about some pipeline running in a server where we connect via SSH? Also, what to do with `plots` command? Do we want to keep the HTML functionality? The plots were made in certain way in order to not have to support our own plotting framework, and use something that probably all of the users share - having browser. I think visualizing `live` using different means will be hard both to maintain and explain to users.

Reply:
Including these features in dvclive doesn't need to impact dvc at all. No need to block this PR or the release.

For users to be able to get value out of dvclive without dvc, could we build a basic summary capability in dvclive? I was wondering whether we could adapt what's currently in https://github.com/iterative/dvc/blob/master/dvc/utils/html.py as a starting point, but it doesn't need to be if there's a simpler way. Looking at that file, it looked like the data structures to feed to `HTML.write()` are reasonably simple, and the data is already being collected by dvclive.

Thoughts? I can open a new issue to discuss if it's worthwhile.